datafusion/functions-nested/src/map_extract.rs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! [`ScalarUDFImpl`] definitions for map_extract functions.
+
+use arrow::array::ArrayRef;
+
+use arrow::datatypes::{DataType, Float64Type, Int64Type, UInt32Type};
+use arrow_array::{
+    new_null_array, Array, ArrowPrimitiveType, MapArray, PrimitiveArray, StringArray,
+};
+use datafusion_common::cast::{as_primitive_array, as_string_array};
+use datafusion_common::utils::get_map_entry_field;
+use datafusion_common::{cast::as_map_array, exec_err, Result};
+use datafusion_expr::{ColumnarValue, ScalarUDFImpl, Signature, Volatility};
+use std::any::Any;
+
+use crate::utils::make_scalar_function;
+
+// Create static instances of ScalarUDFs for each function
+make_udf_expr_and_func!(
+    MapExtract,
+    map_extract,
+    map key,
+    "Return a list containing the value for a given key or an empty list if the key is not contained in the map.",
```

Comment:
Documention is little confusing, from the tests i can see that either we return the value or NULL incase key is absent. Isn't that the case ?

Reply:
I forgot to update this doc. It is resolved now.